test/jdk/java/nio/file/Files/CopyAndMove.java

```diff
@@ -40,14 +40,64 @@
 import java.nio.file.attribute.*;
 import java.util.*;
 import java.util.concurrent.TimeUnit;
+import java.util.stream.Stream;
+import java.util.stream.StreamSupport;
 import jdk.test.lib.Platform;
 import jdk.test.lib.RandomFactory;
 
 public class CopyAndMove {
+    // File system types that support copy-on-write (CoW)
+    // for testing cloning / reflink on Linux
+    private static final Set<String> COW_TYPES =
+        Set.of("bcachefs", "btrfs", "ocfs2", "xfs");
```

Comment:
Shouldn't we also test on non-CoW-FS to be sure that the recent changes do not have a negative impact *there*?

Reply:
I agree about having OS-specific `CopyFile` classes instead of OS-specific providers. I don’t like the `IOStatus` being in the provider and the call of `provider.clone()` seems odd.

- `LinuxNativeDispatcher.ioctl_ficlone` uses `Blocker` emulating the calls in `UnixNativeDispatcher`.
- `FICLONE` is the descendent of `BTRFS_IOC_CLONE` and has the same numerical value so I think it is fairly safe.

Agreed about the other points.

Reply:
Most testing is done on ext4 so this is covered.